tools/testsys/src/run.rs

```diff
@@ -0,0 +1,230 @@
+use crate::aws_resources::AwsK8s;
+use anyhow::{anyhow, ensure, Context, Result};
+use bottlerocket_types::agent_config::TufRepoConfig;
+use buildsys::Variant;
+use clap::Parser;
+use model::test_manager::TestManager;
+use model::SecretName;
+use pubsys_config::InfraConfig;
+use serde::Deserialize;
+use serde_plain::derive_fromstr_from_deserialize;
+use std::collections::HashMap;
+use std::env;
+use std::fs::File;
+use std::path::PathBuf;
+
+/// Run a set of tests on a given arch and variant
+#[derive(Debug, Parser)]
+pub(crate) struct Run {
+    /// The type of test to run
+    test_flavor: TestType,
+
+    /// The arch we to test
```

Comment:
Make sure I'm right about this:

```suggestion
    /// The architecture to test. Either x86_64 or aarch64.
```